packages/babel-core/src/config/plugin.js

```diff
@@ -2,8 +2,8 @@
 
 export default class Plugin {
   constructor(plugin: {}, key?: string) {
-    if (plugin.name != null && typeof plugin.name !== "string") {
-      throw new Error("Plugin .name must be a string, null, or undefined");
+    if (!plugin.name || typeof plugin.name !== "string") {
+      throw new Error("Plugin .name must be a valid JS identifier of type string");
```

Comment:
Happy to just look for name's existence if you think the type check isn't necessary